### ISSUE TITLE
Acc Tests: Possible Role Failures

### ISSUE THREAD

**Issue by jtopjian** [gophercloud/gophercloud#1236]
Recently, some role tests have been failing:

```
=== RUN   TestRoleListAssignmentForGroupOnProject
--- FAIL: TestRoleListAssignmentForGroupOnProject (0.29s)
   identity.go:24: Attempting to create project: ACPTTESTi5W4JkYQ
   identity.go:40: Successfully created project ACPTTESTi5W4JkYQ with ID 385ad2fc17bb44958971f3d3b9d2b904
   identity.go:320: Attempting to find a role
   identity.go:338: Successfully found a role audit with ID 195dff0a483e4539ab8d2dde574c7a91
   identity.go:82: Attempting to create group: ACPTTESTcBWideGh
   convenience.go:35: ESC[1;31mFailure in roles_test.go, line 308: unexpected error ESC[0mESC[1;33m"Internal Server Error"ESC[0mESC[1;31mESC[0m
   identity.go:230: Deleted project: 385ad2fc17bb44958971f3d3b9d2b904
=== RUN   TestRoleListAssignmentForGroupOnDomain
--- FAIL: TestRoleListAssignmentForGroupOnDomain (0.30s)
   identity.go:111: Attempting to create domain: ACPTTESTJInXkOxn
   identity.go:127: Successfully created domain ACPTTESTJInXkOxn with ID 70264e8f1d9f4664ab2457f539900772
   identity.go:320: Attempting to find a role
   identity.go:338: Successfully found a role audit with ID 195dff0a483e4539ab8d2dde574c7a91
   identity.go:82: Attempting to create group: ACPTTESTCcnABYtv
   convenience.go:35: ESC[1;31mFailure in roles_test.go, line 368: unexpected error ESC[0mESC[1;33m"Internal Server Error"ESC[0mESC[1;31mESC[0m
   identity.go:266: Deleted domain: 70264e8f1d9f4664ab2457f539900772
=== RUN   TestRolesAssignToGroupOnDomain
--- FAIL: TestRolesAssignToGroupOnDomain (0.29s)
   identity.go:111: Attempting to create domain: ACPTTESTYQKyjG35
   identity.go:127: Successfully created domain ACPTTESTYQKyjG35 with ID fb9675aa01f7496b9b85e4e2a65d0073
   identity.go:320: Attempting to find a role
   identity.go:338: Successfully found a role audit with ID 195dff0a483e4539ab8d2dde574c7a91
   identity.go:82: Attempting to create group: ACPTTESTFw0Gv4l3
   convenience.go:35: ESC[1;31mFailure in roles_test.go, line 552: unexpected error ESC[0mESC[1;33m"Internal Server Error"ESC[0mESC[1;31mESC[0m
   identity.go:266: Deleted domain: fb9675aa01f7496b9b85e4e2a65d0073
=== RUN   TestRolesAssignToGroupOnProject
--- FAIL: TestRolesAssignToGroupOnProject (0.27s)
   identity.go:24: Attempting to create project: ACPTTESTy286I3Xm
   identity.go:40: Successfully created project ACPTTESTy286I3Xm with ID 72f20596f3ab49f69bd40946b70ae678
   identity.go:320: Attempting to find a role
   identity.go:338: Successfully found a role audit with ID 195dff0a483e4539ab8d2dde574c7a91
   identity.go:82: Attempting to create group: ACPTTESTthMzKR5O
   convenience.go:35: ESC[1;31mFailure in roles_test.go, line 613: unexpected error ESC[0mESC[1;33m"Internal Server Error"ESC[0mESC[1;31mESC[0m
   identity.go:230: Deleted project: 72f20596f3ab49f69bd40946b70ae678
```

I'm not sure if this is a temporary issue or if something has changed upstream.

**Comment by jtopjian** [comment]
Keystone is throwing the following error:

```
    <snip>
    ....
    ref['domain_id'] = cls._get_domain_id_from_token()
AttributeError: type object 'GroupsResource' has no attribute '_get_domain_id_from_token'
```

This is possibly related to Keystone's in-progress move to Flask:

https://github.com/openstack/keystone/commit/27f226554a21667eab62cfc61c13abfc2e780518

At the moment, specifying a Domain ID is required.

**Comment by jtopjian** [comment]
Fixed in #1237
